gcp-firebase/src/main/java/com/baeldung/gcp/firebase/auth/TokenAuthenticationFilter.java

```diff
@@ -0,0 +1,62 @@
+package com.baeldung.gcp.firebase.auth;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseToken;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+@Component
+public class TokenAuthenticationFilter extends OncePerRequestFilter {
+
+    private static final String BEARER_PREFIX = "Bearer ";
+    private static final String USER_ID_CLAIM = "user_id";
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+
+    private final FirebaseAuth firebaseAuth;
+
+    public TokenAuthenticationFilter(FirebaseAuth firebaseAuth) {
+        this.firebaseAuth = firebaseAuth;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws IOException, ServletException {
+        String authorizationHeader = request.getHeader(AUTHORIZATION_HEADER);
+
+        if (authorizationHeader != null && authorizationHeader.startsWith(BEARER_PREFIX)) {
```

Comment:
If there is no Authorization header, do we reject the request? Does the absence of a UsernamePasswordAuthenticationToken in the SecurityContextHolder mean the request will be rejected at a later point?

Reply:
Yes, the default behavior is that if the request was to a secured API endpoint and we don't populate the SecurityContext, then Spring Security returns a 403 with no request body. I've updated the codebase and the article to return custom error response.